pandas/core/indexes/base.py

```diff
@@ -3317,7 +3317,7 @@ def _join_monotonic(self, other, how='left', return_indexers=False):
 
     def _wrap_joined_index(self, joined, other):
         name = self.name if self.name == other.name else None
-        return Index(joined, name=name)
+        return self.__class__(joined, name=name)
```

Comment:
Is there a specific reason you changed this (and the line above that)? (meaning, is there currently something failing due to this?)